File: RedSheet.DbEntities/Models/NanoScopeToNegotiateObjective.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("NanoScopeToNegotiateObjectives", Schema = "dbo")]
    [Serializable]
    public partial class NanoScopeToNegotiateObjective
    {

        [Required]
        [MaxLength(100)]
        public string Buy { get; set; }

        [Required]
        public System.DateTime Date { get; set; }

        [Required]
        [MaxLength(200)]
        public string Focus { get; set; }

        [Required]
        [MaxLength(500)]
        public string KnowAboutThem { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int NanoScopeToNegotiateObjectiveId { get; set; }

        [Required]
        [MaxLength(100)]
        public string OpponentName { get; set; }

        [Required]
        [MaxLength(200)]
        public string Reason { get; set; }

        [RelationshipTableAttribue("NanoRelationshipRequires", "dbo")]
        //Relationships
        public int NanoRelationshipRequireId { get; set; }

        [ForeignKey("NanoRelationshipRequireId")]
        public NanoRelationshipRequire NanoRelationshipRequire { get; set; }

        [RelationshipTableAttribue("NanoScopeToNegotiates", "dbo")]
        //Relationships
        public int NanoScopeToNegotiateId { get; set; }

        [ForeignKey("NanoScopeToNegotiateId")]
        public NanoScopeToNegotiate NanoScopeToNegotiate { get; set; }

        [RelationshipTableAttribue("ProjectModules", "dbo")]
        //Relationships
        public int ProjectModuleId { get; set; }

        [ForeignKey("ProjectModuleId")]
        public ProjectModule ProjectModule { get; set; }

        [RelationshipTableAttribue("ValueObjectives", "dbo")]
        //Relationships
        public int ValueObjectiveId { get; set; }

        [ForeignKey("ValueObjectiveId")]
        public ValueObjective ValueObjective { get; set; }
        [InverseProperty("NanoScopeToNegotiateObjective")]
        public ICollection<NanoOurObjective> NanoOurObjectives { get; set; }

        [InverseProperty("NanoScopeToNegotiateObjective")]
        public ICollection<NanoScopeToNegotiateCommunicationMode> NanoScopeToNegotiateCommunicationModes { get; set; }


        public NanoScopeToNegotiateObjective()
        {
            this.NanoOurObjectives = new HashSet<NanoOurObjective>();

            this.NanoScopeToNegotiateCommunicationModes = new HashSet<NanoScopeToNegotiateCommunicationMode>();

        }

    }
}

File: redsheet-portal/src/app/database-models/nano-scope-to-negotiate-objective.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { NanoRelationshipRequire, NanoScopeToNegotiate, ProjectModule, ValueObjective, NanoOurObjective, NanoScopeToNegotiateCommunicationMode, vNanoScopeToNegotiateObjectiveRecord } from './'
export class NanoScopeToNegotiateObjective {
    constructor(nanoScopeToNegotiateObjective?: NanoScopeToNegotiateObjective | vNanoScopeToNegotiateObjectiveRecord) {
        let properties = ["buy", "date", "focus", "knowAboutThem", "nanoScopeToNegotiateObjectiveId", "opponentName", "reason", "nanoRelationshipRequireId", "nanoScopeToNegotiateId", "projectModuleId", "valueObjectiveId", "nanoOurObjectives", "nanoScopeToNegotiateCommunicationModes",];
        for (let property of properties)
            if (nanoScopeToNegotiateObjective && nanoScopeToNegotiateObjective[property])
                this[property] = nanoScopeToNegotiateObjective[property];
    }

    @required()
    @maxLength(100)
    buy: string = undefined;

    @required()
    date: Date = undefined;

    @required()
    @maxLength(200)
    focus: string = undefined;

    @required()
    @maxLength(500)
    knowAboutThem: string = undefined;

    nanoScopeToNegotiateObjectiveId: number = 0;

    @required()
    @maxLength(100)
    opponentName: string = undefined;

    @required()
    @maxLength(200)
    reason: string = undefined;

    @range(0, 2147483647)
    nanoRelationshipRequireId: number = undefined;
    nanoRelationshipRequire: NanoRelationshipRequire;

    @range(0, 2147483647)
    nanoScopeToNegotiateId: number = undefined;
    nanoScopeToNegotiate: NanoScopeToNegotiate;

    @range(0, 2147483647)
    projectModuleId: number = undefined;
    projectModule: ProjectModule;

    @range(0, 2147483647)
    valueObjectiveId: number = undefined;
    valueObjective: ValueObjective;
    @nested(NanoOurObjective)
    nanoOurObjectives: NanoOurObjective[];

    @nested(NanoScopeToNegotiateCommunicationMode)
    nanoScopeToNegotiateCommunicationModes: NanoScopeToNegotiateCommunicationMode[];



}
